Lib/ClassLibraryCommon/Analytics/LogRecordStreamReader.cs

```diff
@@ -274,7 +274,7 @@ public void EndCurrentRecord()
             }
             else
             {
-                return long.Parse(temp, NumberStyles.None, CultureInfo.InvariantCulture);
+                return Math.Max(0, long.Parse(temp, NumberStyles.AllowLeadingSign, CultureInfo.InvariantCulture));
```

Comment:
Why don't we want to allow negative longs here?

Couldn't we do 
```
return long.Parse(temp, NumberStyles.AllowLeadingSign, CultureInfo.InvariantCulture);
```

Reply:
Yes, this should also work. The only reason to avoid negatives is that previously the framework would throw an exception, so it might be a breaking change for customer.